frappe/public/js/frappe/form/quick_entry.js

```diff
@@ -112,50 +114,60 @@ frappe.ui.form.QuickEntryForm = class QuickEntryForm {
 
 	validate_for_prompt_autoname() {
 		if (this.meta.autoname && this.meta.autoname.toLowerCase() === "prompt") {
-			this.mandatory = [
+			this.docfields = [
 				{
 					fieldname: "__newname",
 					label: __("{0} Name", [__(this.meta.name)]),
 					reqd: 1,
 					fieldtype: "Data",
 				},
-			].concat(this.mandatory);
+			].concat(this.docfields);
 		}
 	}
 
+	setup_script_manager() {
+		this.script_manager = new frappe.ui.form.ScriptManager({
+			frm: this,
+		});
+		this.script_manager.setup();
+	}
+
 	render_dialog() {
 		var me = this;
+
 		this.dialog = new frappe.ui.Dialog({
-			title: __("New {0}", [__(this.doctype)]),
-			fields: this.mandatory,
+			title: this.get_title(),
+			fields: this.docfields,
 			doc: this.doc,
 		});
 
 		this.register_primary_action();
-		!this.force && this.render_edit_in_full_page_link();
-		// ctrl+enter to save
-		this.dialog.wrapper.keydown(function (e) {
-			if ((e.ctrlKey || e.metaKey) && e.which == 13) {
-				if (!frappe.request.ajax_count) {
-					// not already working -- double entry
-					me.dialog.get_primary_btn().trigger("click");
-					e.preventDefault();
-					return false;
-				}
-			}
-		});
+		this.render_edit_in_full_page_link();
+		this.setup_cmd_enter_for_save();
 
 		this.dialog.onhide = () => (frappe.quick_entry = null);
 		this.dialog.show();
 
 		this.dialog.refresh_dependency();
 		this.set_defaults();
 
+		this.script_manager.trigger("refresh");
```

Comment:
Hey @rmehta, I love this PR, but won't this break for QuickEntry of most documents? Indeed, the refresh might try to perform actions that are not possible, such as `frm.set_query`, `frm.add_fetch`, etc.

![](https://github.com/user-attachments/assets/b9fbb2d9-abe2-4acb-8f79-fd44db8e5987)

Reply:
good point, let me check!